src/api/types/index.ts

```diff
@@ -138,5 +138,5 @@ export type DynamicFieldsResponse = {
 
 export type DynamicFieldRequest = Pick<
   DynamicField,
-  "id" | "name" | "options" | "question_type"
+  "is_default" | "name" | "options" | "order" | "question_type" | "role"
```

Comment:
originally thought the put endpoint wouldn't work without id, but I guess you're getting the id from somewhere else?

Reply:
i updated the `putField` arg type to `& { id: number }` :)